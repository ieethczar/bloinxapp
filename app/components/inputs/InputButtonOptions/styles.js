import { StyleSheet } from 'react-native';

import * as theme from '../../../theme';

const styles = StyleSheet.create({
  container: {
    marginBottom: 15,
  },
  label: {
    color: theme.grayText,
    fontSize: 16,
    fontWeight: 'bold',
    color: '#8F9CB3',
    marginBottom: 5,
  },
});

export default styles;