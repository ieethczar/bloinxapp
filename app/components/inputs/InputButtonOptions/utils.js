const determineIndex = (options, value) => {
  const indexValue = options.indexOf(value);
  if(indexValue !== -1){
    return indexValue;
  }
  return 0;
}

export default {
  determineIndex,
};