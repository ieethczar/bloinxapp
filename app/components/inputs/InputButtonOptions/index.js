import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { ButtonGroup, Text } from 'react-native-elements';

import utils from './utils';
import styles from './styles';

const InputButtonOptions = ({ label, options, value, onPress }) => {
  const handleOnPress = (indexSelected) => onPress(options[indexSelected]);
  const selectedIndex = utils.determineIndex(options, value);

  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <ButtonGroup
        buttons={options}
        selectedIndex={selectedIndex}
        onPress={handleOnPress}
      />
    </View>
  )
};

InputButtonOptions.propTypes = {
  label: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  value: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default InputButtonOptions;