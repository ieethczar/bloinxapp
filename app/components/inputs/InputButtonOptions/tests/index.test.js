import React from 'react';
import renderer from 'react-test-renderer';

import InputButtonOptions from '../index';

describe('InputButtonOptions', () => {
  const props = {
    label: "some-label",
    options: ['uno','two','three'],
    value: "one",
    onPress: jest.fn((arg) => arg),
  };

  it('should render correctly', () => {
    const component = renderer.create(
      <InputButtonOptions {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});