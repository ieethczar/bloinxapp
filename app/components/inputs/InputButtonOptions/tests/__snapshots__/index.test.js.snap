// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButtonOptions should render correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#8F9CB3",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    some-label
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 5,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16.714285714285715,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
          >
            uno
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.714285714285715,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
          >
            two
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.714285714285715,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
          >
            three
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
